packages/derived-wallet-solana/src/createSiwsEnvelope.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  createStructuredMessageStatement,
+  createTransactionStatement,
+  StructuredMessage,
+} from '@aptos-labs/derived-wallet-base';
+import { AnyRawTransaction, Hex, HexInput } from '@aptos-labs/ts-sdk';
+import { SolanaSignInInputWithRequiredFields } from '@solana/wallet-standard-util';
+import { PublicKey as SolanaPublicKey } from '@solana/web3.js';
+
+export interface CreateSiwsEnvelopeInput {
+  solanaPublicKey: SolanaPublicKey;
+  signingMessageDigest: HexInput;
+}
+
+function createSiwsEnvelope(input: CreateSiwsEnvelopeInput & {
+  statement: string
+}): SolanaSignInInputWithRequiredFields {
+  const { solanaPublicKey, signingMessageDigest, statement } = input;
+  const digestHex = Hex.fromHexInput(signingMessageDigest).toString();
+  return {
+    address: solanaPublicKey.toString(),
+    domain: window.location.host,
+    uri: window.location.origin,
+    nonce: digestHex,
```

Comment:
using a default value in the `nonce` field is a bit weird from a dev POV, since the `signMessage()` the dev uses requires them to provide `message, nonce` (following the `AptosSignMessageInput` standard)

So we ended up requiring the dev to provide a `nonce` but never really use it and use the `digestHex` by default

Reply:
The user-provided nonce is not displayed, but it's included in the signing message and will affect the digest

The alternative is that we display the user-provided nonce in the `Nonce` field, and the digest somewhere else.

Reply:
> The alternative is that we display the user-provided nonce in the Nonce field, and the digest somewhere else.

We can also not require the end dev to provide a nonce, and have the adapter provider provide the nonce as the transaction hash